src/app/reporting/Engine.cpp

```diff
@@ -259,11 +268,21 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
     chip::System::PacketBufferTLVWriter reportDataWriter;
     ReportDataMessage::Builder reportDataBuilder;
     chip::System::PacketBufferHandle bufHandle = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+    uint16_t reservedSize                      = 0;
 
     VerifyOrExit(!bufHandle.IsNull(), err = CHIP_ERROR_NO_MEMORY);
 
+    if (bufHandle->AvailableDataLength() > kMaxSecureSduLengthBytes)
+    {
+        reservedSize = static_cast<uint16_t>(bufHandle->AvailableDataLength() - kMaxSecureSduLengthBytes);
+    }
+
     reportDataWriter.Init(std::move(bufHandle));
 
+    // Some platforms, the actual buffer capacity might be larger than kMaxSecureSduLengthBytes, then we may write more data in one
```

Comment:
"Always limit the size of the generated packet to fit within an IPv6 MTU regardless of the available buffer capacity".

Reply:
It should be kMaxSecureSduLengthBytes instead of MTU infact